ml/src/main/scala/ai/h2o/sparkling/ml/backend/H2OMetricsHolder.scala

```diff
@@ -17,6 +17,20 @@
 
 package ai.h2o.sparkling.ml.backend
 
-private[sparkling] case class H2OMetricsHolder(trainingMetrics: Map[H2OMetric, Double],
-                                               validationMetrics: Map[H2OMetric, Double],
-                                               crossValidationMetrics: Map[H2OMetric, Double])
+import scala.collection.mutable
+
+
+private[sparkling] case class H2OMetricsHolder(trainingMetrics: TrainingMetrics,
+                                               validationMetrics: ValidationMetrics,
+                                               crossValidationMetrics: CrossValidationMetrics) {
+}
+
+class H2OMetrics(previousMap: Map[H2OMetric, Double]) extends mutable.HashMap[H2OMetric, Double] {
+  this ++= previousMap.toSeq
+}
+
+class TrainingMetrics(previousMap: Map[H2OMetric, Double]) extends H2OMetrics(previousMap)
```

Comment:
What is the purpose of these subclasses? Why `Map[H2OMetric, Double]` wasn't sufficient?

Reply:
Introduced mainly because of tests to verify that the current metrics return correct object. If you have different idea, please let me know :)

Reply:
I assume traning metrics, validation metrics and cross validation metrics are different. I would be possible to test them based on content and according to the implementation of `currentMetrics` it seems that it would be possible bases on references?

Reply:
Good point! I was so focused on this that I even didn't realize this is an option!

Reply:
I'm not really sure why I haven't about this at first:

``assert(model.metrics.trainingMetrics == model.currentMetrics)``

Thank you!